packages/amplify-codegen-appsync-model-plugin/src/visitors/appsync-typescript-visitor.ts

```diff
@@ -83,12 +96,22 @@ export class AppSyncModelTypeScriptVisitor<
       .withName(modelName)
       .export(true);
 
+    let ownerFieldExists = false;
     modelObj.fields.forEach((field: CodeGenField) => {
+      if ("owner" === this.getFieldName(field) && "String" === this.getNativeType(field)) {
```

Comment:
The field name that stores owner value can be changed with `ownerField` in auth directive. Model gen should support overriding this
```gql
type Post
  @model
  @auth(
    rules: [
      { allow: owner, ownerField: "ownedBy", operations: [create, update, delete, read] },
    ])
{
  id: ID!
  title: String!
  # ownedBy: String would be automatically added
}
```
You can find the definition of `auth` directive here
https://docs.amplify.aws/cli/graphql-transformer/directives#definition-2